pkg/fleet/installer/service/datadog_agent.go

```diff
@@ -51,12 +51,12 @@ var (
 )
 
 // SetupAgent installs and starts the agent
-func SetupAgent(ctx context.Context) (err error) {
+func SetupAgent(ctx context.Context, _ []string) (err error) {
 	span, ctx := tracer.StartSpanFromContext(ctx, "setup_agent")
 	defer func() {
 		if err != nil {
 			log.Errorf("Failed to setup agent: %s, reverting", err)
-			RemoveAgent(ctx)
+			err = RemoveAgent(ctx)
```

Comment:
You're going to break the telemetry here: we will get the error of RemoveAgent instead of the error that triggered the rollback

Reply:
yes lgtm